src/cdk-experimental/menu/menu-stack.ts

```diff
@@ -126,3 +126,19 @@ export class MenuStack {
     return this._elements[this._elements.length - 1];
   }
 }
+
+/** NoopMenuStack is a placeholder MenuStack used for inline menus. */
+export class NoopMenuStack extends MenuStack {
```

Comment:
TypeScript lets you do `implements MenuStack` so you can conform to the same API without actually inheriting.

Reply:
That seems to work only if all members are public. With private and protected it won't allow it.

Reply:
Ah, right, you can use TypeScript `keyof` to accomplish this:
https://www.typescriptlang.org/play/#code/MYGwhgzhAECyCmA7ArgZQC5mAa2gbwChpjoAHAJwEsA3MdeaAfUsXvMTBAXQAsB7ACYAKAJT4iJScD6IIfEPAB0IPgHMhAcgAeGkQG4JxAL4FDZZACMQlYNFB8I8UQC5o1PpQH5oJk6fQAnqQMAAqW1sAAgqSUADwAKgB80AC83gDaANLQLNDY8AF8AGbQ8QC6rvFZZT4GBKCQMAByfHykCCgYWLiUALakCr1I6DBhVjbRcR1omDjJhFIqji5uHl4LkiTSsvJKKuoahboGkr5AA